### PR TITLE
Changes to the partitioning of biomass to coarse woody debris classes

### PR DESCRIPTION
**Pull request by adamhb**
### Description:
This pull request is associated with discussion [962](https://github.com/NGEET/fates/discussions/962) which was continued in discussion [989](https://github.com/NGEET/fates/discussions/989).

The changes proposed here update the partitioning of biomass (struct + sapw) to coarse woody debris classes (cwd; aka cwd fuel classes). Fates currently sends some biomass to all fuel classes regardless of cohort dbh. For example it sends stem biomass to the 1,000 hr fuel class (logs > 7.6 cm in diameter) even when the cohort dbh is much less than 7.6 cm. 

The changes proposed here update the cwd partitioning for small cohorts to be based on cohort dbh such that stem biomass is sent to the appropriate cwd class based on the Rothermel fuel class diameter thresholds (see comment from @jkshuman in discussion [962](https://github.com/NGEET/fates/discussions/962). The proportions of stem biomass sent to each flammable cwd class (twig, small branch, and large branch) is kept constant, where possible, to minimize changes in fuel properties. Partitioning for cohorts larger than 7.6 cm in dbh is not changed.

Changes are applied to all subroutines where stem biomass is partitioned into cwd classes: CWDInput (ED physiology), mortality_litter_fluxes (patch dynamics), fire_litter_fluxes (patch dynamics), logging_litter_fluxes (logging), and SendCohortToLitter (cohort dynamics).

### Collaborators:
@lmkueppers @ZacharyRobbins @ckoven @rgknox @jkshuman @jenniferholm

### Expectation of Answer Changes:
Changes should produce more flammable coarse woody debris (twig, small branch, and large branch) and less non-flammable cwd (1,000 hr fuel). These changes should be most apparent a) during early forest development when cohorts are < 7.6 cm in diameter and, b) for pfts that generally stay small (i.e. shrubs). Fuel properties such as SAV and bulk density are calculated as weighted averages of the fuel classes where the weights are the fraction of total fuel belonging to each class. Therefore, when the proportion of these classes change on the landscape, the aggregate fuel properties will also change, thereby changing fire behavior.

### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the in-code documentation .AND. (the [technical note](https://github.com/NGEET/fates-docs) .OR. the wiki) accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/master/CONTRIBUTING.md) document.
- [ ] FATES PASS/FAIL regression tests were run
- [x] If answers were expected to change, evaluation was performed and provided

### Test Results:
I have tested how these changes affect the accumulation of cwd, fuel properties, and fire behavior for a lower montane mixed conifer forest site in California's Sierra Nevada.

CTSM (or) E3SM (specify which) test hash-tag:
ctsm5.1.dev115 (commit#17e2acb6a)

CTSM (or) E3SM (specify which) baseline hash-tag:
ctsm5.1.dev115 (commit#17e2acb6a)

FATES baseline hash-tag:
sci.1.61.0_api.25.0.0 (commit#e663a6e6)

Test Output:

These changes result in a larger accumulation of flammable coarse woody debris classes (twig, small branch, and large branch) and less accumulation of non-flammable cwd (1,000 hr fuel) during early forest development (0-20 simulation yrs). Differences attenuate as the simulation progresses when more of the cwd is coming from larger cohorts (which are not affected by these changes). Although there is more potentially burnable fuel, the landscape is less flammable on any given day for the same FDI because SAV ratio and fuel bulk density move towards less flammable values due to a higher large branch to leaf fuel ratio on the landscape (fuel properties are weighted averages of the classes). These changes to fuel properties result in less burned area on average, and changes to fire line intensity and vary through time. Fire line intensity was the same in some decades and increased in some decades, likely due to more build up of flammable fuels (see slide 8 of results).

Results can be found [here](https://docs.google.com/presentation/d/1rfpmQ1VoIz6LsXRuxWi5wVnJfF7W68WpbdAoA2l4svU/edit#slide=id.g2001a525273_0_11).

### CODE REVIEW

File: biogeochem/FatesLitterMod.F90 [hunk 1]
```diff
@@ -424,6 +424,73 @@ function GetTotalLitterMass(this) result(total_mass)
...
+     
+     real(r8), parameter :: lb_max_diam        = 7.6 !max diameter [cm] for large branch
+     real(r8), parameter :: sb_max_diam        = 2.5 !max diameter [cm] for small branch
+     real(r8), parameter :: twig_max_diam      = 0.6 !max diameter [cm] for twig
+     !------------------------------------------------------------------------------------
+
+     
+     SF_val_CWD_frac_adj = SF_val_CWD_frac
+     
+     !If dbh is larger than max size of a large branch (1,000 hr fuel) then we don't change
```
Comment (ZacharyRobbins) [inline biogeochem/FatesLitterMod.F90:458]: If DBH is larger than the max size of a large branch (and is thus a 1000hr fuel)? Wording here is confusing. 7.6 is the max boundary for 100 hour fuels in Rothermel.  Thank you

File: biogeochem/FatesLitterMod.F90 [hunk 2]
```diff
@@ -424,6 +424,73 @@ function GetTotalLitterMass(this) result(total_mass)
...
+     !how biomass is partitioned among cwd classes.
```
Comment (ZacharyRobbins) [inline biogeochem/FatesLitterMod.F90:459]: For posterity I have checked the math, and the proportions do hold under this redistribution scheme

File: biogeochem/FatesLitterMod.F90 [hunk 3]
```diff
@@ -424,6 +424,73 @@ function GetTotalLitterMass(this) result(total_mass)
...
+     type(integer), intent(in)          :: ncwd !number of cwd pools
+     real(r8), intent(in)               :: SF_val_CWD_frac(:) !fates parameter specifying the
+                                                              !fraction of struct + sapw going
+                                                              !to each CWD class
+     real(r8), intent(out)              :: SF_val_CWD_frac_adj(:) !Updated cwd paritions
+     !
+     !LOCAL VARIABLES
+     !These diameter ranges are based on work by Fosberg et al., 1971 
+     
+     real(r8), parameter :: lb_max_diam        = 7.6 !max diameter [cm] for large branch
```
Comment (ZacharyRobbins) [inline biogeochem/FatesLitterMod.F90:450]: Please add fuel classes next to comments on size to increase ability to cross reference with Fosberg and Rothermel.

**Review by ZacharyRobbins — CHANGES_REQUESTED** [review]
Feel free to email me with any questions!

File: biogeochem/FatesLitterMod.F90 (hunk 2 above)
Comment (adamhb) [reply biogeochem/FatesLitterMod.F90:459]: Cool, thanks!

Comment (adamhb) [reply biogeochem/FatesLitterMod.F90:459]: PS I somehow made your other two comments disappear, but the code here now has the clarifying points in the comments you requested.

**Review by ZacharyRobbins — APPROVED** [review]

**Comment by adamhb** [comment]
Thanks Zachary for suggesting the decadal-scale figures which can be found on slide 8 of the results [here](https://docs.google.com/presentation/d/1rfpmQ1VoIz6LsXRuxWi5wVnJfF7W68WpbdAoA2l4svU/edit#slide=id.g23d17ff7f5a_0_0)

A new insight from these figures is that mean fire line intensity increases in most decades with the proposed changes. This is likely due to having more combustible fuel on the landscape.

**Review by rgknox — APPROVED** [review]
looks good

**Comment by glemieux** [comment]
Regression testing on cheyenne is complete.  All expected tests pass with multiple `DIFFs`.  The set of differing variables appear to make sense given the changes made here (i.e. variables related to fire and litter in testmods where they are reported). 

Folder location: `/glade/u/home/glemieux/scratch/ctsm-tests/tests_pr1008`

**Pull request merged**